### ISSUE TITLE
[BUG] QZ not working with Echelon Stride

### ISSUE THREAD

**Issue by speedf1** [cagnulein/qdomyos-zwift#1644]
**Describe the bug**
I can't get the Echelon Stride to work with the QZ app.

I added just the  Speed, Inclination, Odometer and Time To Next tiles.  However, only Speed, Odometer and Time To Next show.  Although QA connects to the stride (stride shows APP), the tiles don't seem to reflect the status of the stride.

Debug log and video attached.

https://github.com/cagnulein/qdomyos-zwift/assets/87641705/704c1b6d-c244-4cf9-a0b4-a1e0c1507724

[debug-Tue_Sep_12_14_23_42_2023.log](https://github.com/cagnulein/qdomyos-zwift/files/12586954/debug-Tue_Sep_12_14_23_42_2023.log)

**Comment by cagnulein** [comment]
Hi! The main issue is that QZ is connected to a bike instead of a treadmill.
Is the ECH-FR-000021 your treadmill?
If so I have to do a small modification to make it work
Let me know!

Roberto Viola
Software engineer and open source enthusiast
http://robertoviola.cloud



Il giorno mar 12 set 2023 alle ore 15:39 speedf1 ***@***.***>
ha scritto:

**Comment by speedf1** [comment]
Hi, yes ECH-FR-00021 is the treadmill.  I do also have a echelon sport bike (ECH-SPORT-021425) as well though.

**Comment by cagnulein** [comment]
Ok modification done, I already added to the beta. Click here to join https://play.google.com/apps/testing/org.cagnulen.qdomyoszwift
You have to see the version 2.16.11 with the fix! It will be online in the next hours
Let me know!

**Comment by speedf1** [comment]
Hi @cagnulein , no sign of it yet on Google Play; do you know if it has been released?

**Comment by cagnulein** [comment]
yes it's already out. Did you click on the link above?

**Comment by speedf1** [comment]
Yup.  Google Play store shows that I am a beta tester.  I uninstalled the app, waiting a few minutes, went back into play store and downloaded, but I'm still only getting version 2.16.7 downloaded to my device.

![Screenshot_2023-09-13-14-39-22-63_b5a5c5cb02ca09c784c5d88160e2ec24](https://github.com/cagnulein/qdomyos-zwift/assets/87641705/8abc02a7-8934-4c32-8380-d4337759a064)

**Comment by cagnulein** [comment]
is your google play account equal to the gmail address from which you wrote me?

**Comment by speedf1** [comment]
Yes. k*****1@googlemail.com

**Comment by cagnulein** [comment]
yes I double checked it. If you want you can download the trial from here if you want to check that the solutions is working meanwhile google is updating it (but other users already installed it, so I know for sure that it's online)

https://github.com/cagnulein/qdomyos-zwift/suites/16082030662/artifacts/918095291

**Comment by speedf1** [comment]
Thanks.  I've downloaded the apk but probably won't have a chance to test today.

**Comment by cagnulein** [comment]
No problem, take your time.

Il giorno mer 13 set 2023 alle 16:23 speedf1 ***@***.***> ha
scritto:

**Comment by speedf1** [comment]
So a quick test of the beta app seems to show it correctly working with speed and elevation.   However, because I had to uninstall the old app to install the new one, I lost all my settings.  For some reason, QZ can't now connect to my peloton account ("Peloton Login Error!").  I thought QZ just needed my email address and password entered into the appropriate Peloton fields in the app, but that doesn't seem to work anymore; thoughts, or what am I doing wrong?

**Comment by cagnulein** [comment]
yeah that's an issue of trial version, let's install back the original one from the store and let's wait until google will review it

**Comment by speedf1** [comment]
Oh no!  I've just spend ages setting up a couple of profiles (one for the bike and one for the tread).  How can I save those?  When I tried to save (from the non-beta version) and restore (into the beta), the profiles didn't restore.

**Comment by cagnulein** [comment]
do you still have the files? if so just backup them on a local directory
and put them again in the qz folder after
sorry but i wasn't aware of your efforts on this
i thought you just downloaded qz the last day

Il giorno gio 14 set 2023 alle 17:36 speedf1 ***@***.***> ha
scritto:

**Comment by speedf1** [comment]
No problem all.  Still love the app!  Definitely going to donate something for this support!  So do I go back to the non-beta version in the app store and wait, or just delete the apk version you sent and wait for an official beta on the app store?

**Comment by cagnulein** [comment]
i guess in order to return to the store one you need to download the actual
installed.
but be sure to check if you still have the profiles in the qz directory

Il giorno gio 14 set 2023 alle 17:43 speedf1 ***@***.***> ha
scritto:

**Comment by cagnulein** [comment]
let's bypass google, click here https://play.google.com/apps/internaltest/4698920315311114696

**Comment by speedf1** [comment]
Thanks, but I think all sorted.  I deleted the QZ Beta apk you sent, and reinstalled from google play.  That seems to have downloaded the beta but peloton login now works.  I used a file manager to save the files in the "profiles" directory of QZ and restored them using the file manager once re-downloaded and that looks fine.

Only thing now is that this beta doesn't seem to be able to control the speed/incline of the treadmill by pressing the +/- buttons, even though "Treadmill speed forcing" is switch on.  Also looks like the "Pause when App Starts", when switched off the  treadmill still doesn't start until you press the start button in Qz
